### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by lyw0149**
1. src/server -> dist-server/
2. src/server & src/test -> dist-test/
3. run lint & test automatically before `git push`
4. refactor gulp task 

@nlpenguin @ethkim @djflexible  @ifndefdeadmau5 @SarahhhC @KaylaKwon  @festinalente91

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
   "name": "yoda",
   "version": "1.0.0",
-  "description": "This is skeleton of project yoda.",
+  "description": "Yoda Sever.",
   "main": "server.js",
   "scripts": {
     "test": "gulp test",
-    "start": "gulp"
+    "start": "gulp",
+    "jscs": "jscs"
```
Comment (ethkim) [inline package.json]: This is awesome! Thank you!!

File: src/server/controllers/users.controller.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ export function getAll(req, res, next) {
 // Get all user list except logged in user
 export function getMentorList(req, res, next) {
   // TODO: Once session_auth is implemented, replace the email address with req.session.email.
-  User.find({ 'email' : { $ne : 'session@yoda.com' }},
+  User.find({ email: { $ne: 'session@yoda.com' } },
```
Comment (ethkim) [inline src/server/controllers/users.controller.js]: You'll have a conflict with @SarahhhC 's PR. Please ensure to update before submit (whoever submits after :)

File: src/server/controllers/users.controller.js [hunk 3]
```diff
@@ -92,6 +92,7 @@ export function signin(req, res, next) {
         res.status(400).json(err);
       } else {
         if (doc.length == 0) {
+
```
Comment (ethkim) [inline src/server/controllers/users.controller.js]: Why was this line added?

File: src/server/controllers/users.controller.js [hunk 4]
```diff
@@ -92,6 +92,7 @@ export function signin(req, res, next) {
         res.status(400).json(err);
       } else {
         if (doc.length == 0) {
```
Comment (ethkim) [inline src/server/controllers/users.controller.js]: I think we need to start using ===.
https://github.com/airbnb/javascript#comparison--eqeqeq

File: src/test/utils.js [hunk 5]
```diff
@@ -15,12 +15,13 @@ beforeEach(done => {
...
     return done();
   }
-  
+
   if (mongoose.connection.readyState === 0) {
     mongoose.connect('mongodb://localhost:27017/yoda', function (err) {
       if (err) {
         throw err;
       }
+
```
Comment (ethkim) [inline src/test/utils.js]: Remove this line.

**Comment by ethkim** [comment]
This is an awesome change! Our codebase is getting into a much better shape. :-) 👍

**Comment by ethkim** [comment]
Hmm.. I thought I set .lgtm config to have :+1: approve PRs. LGTM (modulo the comments above).

**Comment by ifndefdeadmau5** [comment]
LGTM
